Filter

ajit-sharmaGH opened the pull request:
![image](https://user-images.githubusercontent.com/80311905/161705364-667dca88-a08b-4846-a147-a90db729990a.png)

ajit-sharmaGH committed:
did some minor changes

ajit-sharmaGH committed:
feat: added products filter functinality
- range pricing
- category filter
- rating filter
- sort by pricing

mariyasada commented on src/components/cart/cartNav.js:
```diff
@@ -0,0 +1,86 @@
+          </li>  
+          <li className="badge_container">
+            <Link to="/cart">
+              <i className="fa-solid fa-cart-arrow-down"> </i>  
+             
+            </Link>  
+            <span className="icon-badge"> 4 </span>  
+          </li>  
+          <li className="badge_container">
+            <Link to="/wishlist">
+              <i className="fa-solid fa-heart"> </i>  
+             
+            </Link>  
+            <span className="icon-badge"> 2 </span>  
```
remove unnecessary space

mariyasada commented on src/components/productsFilter/productsFilter.jsx:
```diff
@@ -1,33 +1,143 @@
+            Max={10000}
+            Step={150}
+            defaultValue={state.range}
+            onChange={
+              
+                (event) => dispatch({ type: "RANGE", payload: event.target.value })
+            }
```
same for this

mariyasada commented on src/context/filter-context.js:
```diff
@@ -0,0 +1,22 @@
+
+  return (
+    <ProductsFilterContext.Provider value={{ state, dispatch }}>
+      {" "}
+      {children}{" "}
+    </ProductsFilterContext.Provider>
+  );
```
remove this empty braces

mariyasada commented on src/index.js:
```diff
@@ -4,12 +4,19 @@ import "./index.css";
       <App />
+     
+      </FilterContextProvider>
+    
```
remove spaces

mariyasada reviewed (approved):
The code structure is good. just remove unnecessary space

ankit9015 commented on src/components/cart/cartNav.js:
```diff
@@ -0,0 +1,86 @@
+import { Link } from "react-router-dom";
```
One suggestion, you can have one Navigation header for all pages. Since there is not much change in the navigation component across different pages.

ankit9015 commented on src/components/cart/cartNav.js:
```diff
@@ -0,0 +1,86 @@
+      >
+        <ul className="sidebar-navlink">
+          <h1 className="sidebar-logo"> Lifestyle Store </h1>
+          <li onClick={showSidebar}>
```
Like how you are displaying sidebar when the button is clicked

ankit9015 commented on src/components/products/productsNav.js:
```diff
@@ -1,50 +1,53 @@
   return (
-    <div>
+    <>
+   
```
This navigation header is mostly the same as the one for the cart page, hence you can create one global navigation component and place it in App.js

ankit9015 commented on src/index.js:
```diff
@@ -4,12 +4,19 @@ import "./index.css";
 makeServer();
 ReactDOM.render(
   <React.StrictMode>
+   <FilterContextProvider>
```
Extra spaces can be removed

ankit9015 commented on src/reducers/productsFilterReducer.js:
```diff
@@ -0,0 +1,82 @@
+    search:'',
+    rating: 1,
+    range: 1000,
+
```
here also, spaces can be removed

ankit9015 commented on src/reducers/productsFilterReducer.js:
```diff
@@ -0,0 +1,82 @@
+
+const getFilteredProducts = (
+    products,
+
```
same

ankit9015 reviewed (approved):
PR is good to merge.

ajit-sharmaGH committed:
fixed reviewed changes
- created a separate navbar component for all elements
- removed some spaces and formatted the code
- added Link tag to home page category selection

ajit-sharmaGH commented:
thank you @mariyasada and @ankit9015 for the suggestions, resolved !
